vendor/github.com/docker/distribution/configuration/parser.go

```diff
@@ -220,7 +220,7 @@ func (p *Parser) overwriteStruct(v reflect.Value, fullpath string, path []string
 		}
 	case reflect.Ptr:
 		if field.IsNil() {
-			field.Set(reflect.New(field.Type().Elem()))
+			field.Set(reflect.New(sf.Type))
```

Comment:
i'm not sure why this change is being picked up unless we were carrying a patch in our vendor dir that is not in openshift/docker-distribution ?

Reply:
it looks like the change came from 7746110b40a1689628d79c8ba35822746ffd3160 

so we need to get 7746110b40a1689628d79c8ba35822746ffd3160 into openshift/docker-distribution#image-registry-release-2.6.2

Reply:
pr here: https://github.com/openshift/docker-distribution/pull/2

Reply:
> i'm not sure why this change is being picked up unless we were carrying a patch in our vendor dir that is not in openshift/docker-distribution ?

@mfojtik told me that changes in the `vendor` directory should be made in our repository and then transferred to the `openshift/docker-distribution` repository by some script.

Before that, I was opening PR: https://github.com/openshift/docker-distribution/pull/1

Reply:
well barring more details on who's implementing that script, i'm going to proceed with the manual approach.